File: packages/java/src/main/java/com/designpatterns/PatternCategory.java

package com.designpatterns;

public enum PatternCategory {
    CREATIONAL("Creational"),
    STRUCTURAL("Structural"),
    BEHAVIORAL("Behavioral");

    private final String value;

    PatternCategory(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: packages/nuget/src/PatternCategory.cs

using System.Collections.Generic;

namespace DesignPatterns.Catalog;

public static class PatternCategory
{
    public const string Creational = "Creational";
    public const string Structural = "Structural";
    public const string Behavioral = "Behavioral";

    public static readonly IReadOnlyList<string> All = new[]
    {
        Creational,
        Structural,
        Behavioral,
    };
}
